RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -502,7 +502,7 @@ bool DAClusterizerInZ_vect::merge(vertex_t& y, track_t& tks, double& beta) const
     }
 #endif
 
-    if (rho > 0) {
+    if (rho > 1.e-100) {
       y.zvtx[k] = (y.rho[k] * y.zvtx[k] + y.rho[k + 1] * y.zvtx[k + 1]) / rho;
```

Comment:
I'm not sure this cut off is good here.
PV reco often has some convergence issues.
It would make sense to me to cut here if the result is bad.
1e100 is pretty far away from 1e308 (max double).
Presumably both `rho_k` and `rho_k+1` are non-negative here. The result should be mostly finite here.

Is it possible to check the result for finiteness instead?
In fact, in this case we can even use the CMS context and cut at something like `std::abs(y.zvtx[k])<1e4` (100 m is plenty safe for a PV reco)

Reply:
@AdrianoDee , will you look in to it?

Reply:
I would agree with @slava77 . Here and in the there cases wouldn't it be safer to check could we use direct `std:isfinite` or `std::isnan` ore something similar? Here indeed 1e4 would also make the job. Since the `rho>0` was (ineffectively) there to avoid 0 divisions maybe this should moved to something like:

``` cpp
if (rho > 0) 
{
  y.zvtx[k] = (y.rho[k] * y.zvtx[k] + y.rho[k + 1] * y.zvtx[k + 1]) / rho;
  if(not std::isfinite(y.zvtx[k])) 
     y.zvtx[k] = 0.5 * (y.zvtx[k] + y.zvtx[k + 1]);
} else {
  y.zvtx[k] = 0.5 * (y.zvtx[k] + y.zvtx[k + 1]);
}
```

?